### PR TITLE
Added conditional check tests

### PR DESCRIPTION
**Pull request by praveen4g0**
Added condition check Test on pipeline Run resources
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior

```

### CODE REVIEW

File: test/conditional_check_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(t, c, namespace) }, t.Logf)
+	defer tearDown(t, c, namespace)
+
+	condResource := tb.Condition("file-exists", namespace,
+		tb.ConditionSpec(
+			tb.ConditionParamSpec("path", v1alpha1.ParamTypeString),
+			tb.ConditionResource("workspace", v1alpha1.PipelineResourceTypeGit),
+			tb.ConditionSpecCheck("", "alpine"),
+			tb.ConditionSpecCheckScript("cat $(resources.workspace.path)/$(params.path)| grep \"Cloud Native\""),
```
Comment (sbwsg) [inline test/conditional_check_test.go]: I dunno about this bit - we're kind of assuming that README.md will always contain the words "Cloud Native".  Changing a README file in the future shouldn't result in a e2e test failure and will be super confusing if we ever do.

How about checking whether this very test file exists?  If you want to keep the "grep" then you could add a comment here like `// This comment is used by this test` and then grep for that with `cat $(resources.workspace.path)/$(params.path) | grep \"This comment is used by this test\"`

**Comment by bobcatfish** [comment]
Hey @praveen4g0 ! Thanks for taking the time to add this :D

Is there an error you ran into that motivated you to add this?

I'm wondering b/c one thing that's odd about the end to end tests is that we actually have our tests split across 2 locations:
* The examples folder <-- anything that we can verify by writing and executing a pipeline/task directly we add here (basically the Run will fail if there is a bug
* The end to end go tests <-- we add coverage for things we can't easily express just with the examples

In the case of conditionals we have these 2 examples which test them:
* https://github.com/tektoncd/pipeline/blob/master/examples/pipelineruns/conditional-pipelinerun.yaml
* https://github.com/tektoncd/pipeline/blob/master/examples/pipelineruns/conditional-pipelinerun-with-optional-resources.yaml

So I'm wondering if maybe we don't need to add another end to end test after all?

**Comment by bobcatfish** [comment]
Closing this for now but please reopen if you feel strongly that we should have another test here!

**Pull request closed without merging**